orangecrab/src/ConfigGen.hs

```diff
@@ -1,95 +1,272 @@
+  ilaProbe' :: (Vec n GenSignal, Signal dom a) -> (Vec n GenSignal, Signal dom a)
+  ilaProbe' acc = acc
 
-writeConfig ::
-  forall dom a n ts.
-  ( KnownNat n
-  , NamedSignal ts
-  , Lift a
+-- | Induction case
+instance
+  {-# OVERLAPPABLE #-}
+  ( LabelledSignals cont (n + 1) dom nextS
+  , BitPack a
+  , nextS ~ (s, a)
   ) =>
-  -- | The amount of samples it can store in its buffer
-  SNat n ->
-  -- | Merely an identifier, recommended to be the name of the toplevel design, but it can be any name you fancy
-  -- Used as the toplevel name in the VCD file
-  String ->
-  -- | A list of signals and labels (Strings) tupled together to sample
-  HList ts ->
-  -- | The same list of signals, but then bundled together
-  Signal dom a ->
-  -- | a `IlaConfig` in AST form, this should be passed directly to the ILA
-  IO (Exp)
-writeConfig size toplevelName namedSignals bundledSignals = do
-  let
-    sizeNames = getSizeAndName namedSignals
-
-    genSignals = (uncurry $ flip GenSignal) <$> sizeNames
-
-    nonHashed = GenIla toplevelName (natToNum @n) 0 genSignals
-    hashValue = hash nonHashed
-    hashValueBv = resize $ pack hashValue :: BitVector 32
-    hashedIla = GenIla toplevelName (natToNum @n) hashValue genSignals
-
-    genIlas = GenIlas [hashedIla]
-
-  encodeFile "ilaconf.json" genIlas
-
-  let ilaConf =
-        IlaConfig
-          { hash = hashValueBv
-          , size = size
-          , tracing = bundledSignals
+  LabelledSignals ((Signal dom a, String) -> cont) n dom s
+  where
+  ilaProbe' :: (Vec n GenSignal, Signal dom s) -> (Signal dom a, String) -> cont
+  ilaProbe' (acc, sigAcc) (sig, label) =
+    let
+      newAcc :: Vec (n + 1) GenSignal
+      newAcc =
+        GenSignal
+          { name = label
+          , width = natToNum @(BitSize a)
           }
+          :> acc
+
+      newSig :: Signal dom nextS
+      newSig = bundle (sigAcc, sig)
+     in
+      ilaProbe' (newAcc, newSig)
+
+instance
+  ( LabelledSignals cont 1 dom a
+  , BitPack a
+  , nextS ~ a
+  ) =>
+  LabelledSignals ((Signal dom a, String) -> cont) 0 dom s
+  where
+  ilaProbe' :: (Vec n GenSignal, Signal dom s) -> (Signal dom a, String) -> cont
+  ilaProbe' (Nil, _) (sig, label) = ilaProbe' (newAcc, sig)
+   where
+    newAcc :: Vec 1 GenSignal
+    newAcc =
+      GenSignal
+        { name = label
+        , width = natToNum @(BitSize a)
+        }
+        :> Nil
+
+{- | A polyvariadic function containing 'labelled signals', aka, a list of tuples where the left
+side is an arbitary signal, and the right a string. 
+
+# Example:
+
+>>> counter = register 0 $ counter + 1 :: Signal dom (Unsigned 8)
+>>> active = pure True :: Signal dom Bool
+>>> ilaProbe (counter, "8 bit value") (active "system active")
+-}
+ilaProbe :: (LabelledSignals ((Signal dom a, b) -> t) 0 dom a) => (Signal dom a, b) -> t
+ilaProbe first@(f, _) = ilaProbe' (Nil, f) first
+
+-- | Write signal information to a file, using blackboxes
+writeSignalInfo ::
```

Comment:
This is very aspecific. Please say a bit more about what this function achieves and where it should be used.